.drone.yml

```diff
@@ -61,3 +61,138 @@ steps:
     commands:
       - export CHANGED_FILES="$(git diff --name-status ${DRONE_COMMIT_BEFORE}..${DRONE_COMMIT_AFTER} | grep ^A | cut -f2)"
       - /check-php-cs.sh
+---
+kind: pipeline
+type: docker
+name: php7.3-mariadb
+
+steps:
+  - name: Test Friendica
+    image: friendicaci/php7.3:php7.3.28
+    environment:
+      MYSQL_HOST: "mariadb"
+      MYSQL_PORT: 3306
+      MYSQL_DATABASE: "test"
+      MYSQL_PASSWORD: "test"
+      MYSQL_USER: "test"
+      REDIS_HOST: "redis"
+      MEMCACHED_HOST: "memcached"
+      MEMCACHE_HOST: "memcached"
+    commands:
+      - composer validate
+      - composer install --prefer-dist
+      - cp config/local-sample.config.php config/local.config.php
+      - while ! mysqladmin ping -h$MYSQL_HOST -P$MYSQL_PORT --silent; do sleep 1; done
+      - mysql -h$MYSQL_HOST -P$MYSQL_PORT -p$MYSQL_PASSWORD -u$MYSQL_USER $MYSQL_DATABASE < database.sql
+      - phpunit --configuration tests/phpunit.xml
+
+services:
+  - name: mariadb
+    image: mariadb:latest
+    environment:
+      MYSQL_ALLOW_EMPTY_PASSWORD: "true"
+      MYSQL_DATABASE: "test"
+      MYSQL_PASSWORD: "test"
+      MYSQL_USER: "test"
+    tmpfs:
+      - /var/lib/mysql
+
+  - name: memcached
+    image: memcached
+
+  - name: redis
+    image: redis
+---
+kind: pipeline
+type: docker
+name: php7.4-mariadb
+
+steps:
+  - name: Test Friendica
+    image: friendicaci/php7.4:php7.4.18
+    environment:
+      MYSQL_HOST: "mariadb"
+      MYSQL_PORT: 3306
+      MYSQL_DATABASE: "test"
+      MYSQL_PASSWORD: "test"
+      MYSQL_USER: "test"
+      REDIS_HOST: "redis"
+      MEMCACHED_HOST: "memcached"
+      MEMCACHE_HOST: "memcached"
+      XDEBUG_MODE: "coverage"
+    commands:
+      - phpenmod xdebug
+      - composer validate
+      - composer install --prefer-dist
+      - cp config/local-sample.config.php config/local.config.php
+      - while ! mysqladmin ping -h$MYSQL_HOST -P$MYSQL_PORT --silent; do sleep 1; done
```

Comment:
Isn't this a recipe for forever lock in case MySQL doesn't start properly for any reason?

Reply:
ah I already created a more elegant way, totally forgot about that, thanks :D

Reply:
https://github.com/friendica/friendica/pull/10251/commits/fd08877211563684a45643206badee74149b5903 <-- failed correctly because of wrong `$MYSQL_HOST`
https://github.com/friendica/friendica/pull/10251/commits/7af071ae815e062d0ca71ef254471c2ba21e54b1 <-- fixed it

So the timeout works :-)